webapp/channels/src/components/admin_console/custom_plugin_settings/index.ts

```diff
@@ -83,7 +83,52 @@ function makeGetPluginSchema() {
                         component,
                         showTitle: customComponents[key] ? customComponents[key].options.showTitle : false,
                     };
-                }) as Array<Partial<AdminDefinitionSetting>>;
+                });
+            };
+
+            const parsePluginSettingSections = (sections: PluginSettingSection[]) => {
+                return sections.map((section) => {
+                    const key = section.key.toLowerCase();
+                    let component;
+                    let settings: Array<Partial<AdminDefinitionSetting>> = [];
+                    if (section.custom) {
+                        if (customSections[key]) {
+                            component = customSections[key]?.component;
+                        } else {
+                            // Show warning banner for custom sections when the plugin is disabled.
+                            settings = [{
+                                type: Constants.SettingsTypes.TYPE_BANNER,
+                                label: defineMessage({
+                                    id: 'admin.plugin.customSection.pluginDisabledWarning',
+                                    defaultMessage: 'In order to view this section, enable the plugin and click Save.',
+                                }),
+                                banner_type: 'warning',
+                            }];
+                        }
+                    }
+
+                    if (settings.length === 0) {
+                        settings = parsePluginSettings(section.settings) as Array<Partial<AdminDefinitionSetting>>;
+                    }
+
+                    return {
+                        key,
+                        title: section.title,
+                        subtitle: section.subtitle,
+                        settings,
+                        header: section.header,
+                        footer: section.footer,
+                        component,
+                    };
```

Comment:
I think if you type hint this return (and/or the function sig, depending on ts), then you won't need the cast below (probably same for `parsePluginSettings`)

Reply:
I suffered a bit to make it pass, let me check but no promises :p

Reply:
I think it's working, if that's what you meant.

Reply:
It's more of a desire to avoid the `as` cast, because that will erase any type issues that you'd want to catch (in the future).

Reply:
I know, but this is a mess of union types combined with unknown inputs.